src/client.js

```diff
@@ -230,10 +249,15 @@ class Client extends EventEmitter {
   }
 
   write (name, params) {
-    if (!this.serializer.writable) { return }
-    debug('writing packet ' + this.state + '.' + name)
+    if (!this.customCommunication && !this.serializer.writable) { return }
+    debug(`[${this.state}] from ${this.isServer ? 'server' : 'client'}: ` + name)
     debug(params)
-    this.serializer.write({ name, params })
+
+    if (this.customCommunication) {
+      this.customCommunication.sendData({ name, params, state: this.state })
```

Comment:
It doesn't support writeBundle and writeRaw as I'm not sure how flying-squid could use it